src/main/java/com/teenthofabud/wizard/nandifoods/wms/settings/unit/uom/service/impl/UOMServiceImpl.java

```diff
@@ -361,31 +363,90 @@ public void updateExistingUOMByCode(String code, UOMDtoV2 sourceUOMDto) {
     }
 
     private UOMEntity comparativelyUpdateMandatoryCollection(UOMDtoV2 old, UOMDtoV2 _new, UOMEntity target) {
-        if(old.getLinkedUOMs().isPresent() && _new.getLinkedUOMs().isEmpty()) {
+        target = updateLinkedUOMs(old, _new, target);
+        target = updateMeasuredValues(old,_new,target);
+        return target;
+    }
+
+    private UOMEntity updateMeasuredValues(UOMDtoV2 old,UOMDtoV2 _new, UOMEntity target) {
+        List<UnitClassMeasuredValuesDtoV2> oldMeasuredValues = old.getMeasuredValues();
+        List<UnitClassMeasuredValuesDtoV2> newMeasuredValues = _new.getMeasuredValues();
+        oldMeasuredValues.sort((a,b)->a.getMeasurementSystem().getOrdinal() < b.getMeasurementSystem().getOrdinal() ? -1 : 1);
+        newMeasuredValues.sort((a,b)->a.getMeasurementSystem().getOrdinal() < b.getMeasurementSystem().getOrdinal() ? -1 : 1);
+        target.getMeasuredValues().sort((a,b)->a.getMeasurementSystem().getOrdinal() < b.getMeasurementSystem().getOrdinal() ? -1 : 1);
+        Diff diff = javers.compareCollections(oldMeasuredValues,newMeasuredValues,UnitClassMeasuredValuesDtoV2.class);
+        log.debug("measured values diff : {}",diff.prettyPrint());
+        diff.getChangesByType(PropertyChange.class).forEach(Errors.rethrow().wrap(p -> {
+            log.debug("{} changed from {} to {}", p.getPropertyNameWithPath(), p.getLeft(), p.getRight());
+            String propPath = p.getPropertyNameWithPath();
+            int currentIndex = Integer.parseInt(propPath.substring(propPath.indexOf("/")+1,propPath.indexOf("/")+2));
+            beanUtilsBean.copyProperty(target.getMeasuredValues().get(currentIndex),p.getPropertyName(),p.getRight());
```

Comment:
can't you use bean utils to replace elements in collection directly instead of replacing them individually by their indices?